data/random-teams.ts

```diff
@@ -874,21 +877,23 @@ export class RandomTeams {
 		case 'toxicspikes':
 			return {cull: counter.setupType || teamDetails.toxicSpikes};
 		case 'uturn':
-			const bugAndRecovery = hasType['Bug'] && counter.recovery;
-			return {cull: counter.speedsetup || (counter.setupType && !bugAndRecovery) || (isDoubles && hasMove['leechlife'])};
+			const bugSwordsDanceCase = hasType['Bug'] && counter.recovery && hasMove['swordsdance'];
+			return {cull: counter.speedsetup || (counter.setupType && !bugSwordsDanceCase) || (isDoubles && hasMove['leechlife'])};
 
 		// Ineffective to have two particular moves together
 		case 'explosion':
-			// Special case for Gigalith
-			// [15:33] A Cake Wearing A Hat: meant to prevent rock blast as only stab on choice band
+			// Rock Blast: Special case for Gigalith to prevent Stone Edge-less Choice Band sets
 			const otherMoves = ['curse', 'drainpunch', 'rockblast', 'painsplit', 'wish'].some(m => hasMove[m]);
 			return {cull: counter.speedsetup || counter.recovery || otherMoves};
 		case 'facade':
+			// Special cases for Braviary and regular Snorlax, respectively
 			return {cull: counter.recovery || movePool.includes('doubleedge')};
 		case 'quickattack':
-			const uturnCullCondition = counter.Physical > 3 && movePool.includes('uturn');
-			return {cull: counter.speedsetup || (hasType['Rock'] && counter.Status) || uturnCullCondition};
+			// Diggersby wants U-turn on Choiced sets
+			const diggersbyCull = counter.Physical > 3 && movePool.includes('uturn');
+			return {cull: counter.speedsetup || (hasType['Rock'] && counter.Status) || diggersbyCull};
 		case 'blazekick':
+			// Special case for Genesect. Remove when Genesect changes
 			return {cull: counter.Special >= 1};
```

Comment:
```suggestion
			return {cull: species.id === 'genesect' && counter.Special >= 1};
```
There's nothing inherently wrong with Blaze Kick on a mixed moveset, we just don't want it for Genesect.